scripts/virtualized_install.sh

```diff
@@ -0,0 +1,481 @@
+#!/bin/bash
+set -euo pipefail
+
+SCRIPT_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+source "$SCRIPT_DIR/test_install.sh"
+source "$SCRIPT_DIR/util.sh"
+
+# Configuration for the virtualized installation
+declare -A CONFIG
+
+# Initialize configuration with default values and parameter processing
+function init_virtualized_config() {
+    local distro="${1:-}"
+    local api_domain="${2:-}"
+    local app_domain="${3:-}"
+    local proxy_url="${4:-}"
+    local app_port="${5:-}"
+    local api_port="${6:-}"
+    local email="${7:-}"
+    local password="${8:-}"
+    local env="${9:-}"
+    local show_in_console="${10:-}"
+    local internal_app_proxy_port="${11:-}"
+    local internal_api_proxy_port="${12:-}"
+    
+    [[ -z "$internal_app_proxy_port" ]] && internal_app_proxy_port="7443"
+    [[ -z "$internal_api_proxy_port" ]] && internal_api_proxy_port="8443"
+    [[ -z "$distro" ]] && distro="debian/11"
+    [[ -z "$proxy_url" ]] && proxy_url="127.0.0.1"
+    [[ -z "$app_port" ]] && app_port=$(generate_random_available_port)
+    [[ -z "$api_port" ]] && api_port=$(generate_random_available_port)
+    [[ -z "$env" ]] && env="production"
```

Comment:
```suggestion
    [[ -z "$app_port" ]] && app_port="$(generate_random_available_port)"
    [[ -z "$api_port" ]] && api_port="$(generate_random_available_port)"
    [[ -z "$env" ]] && env="production"
```

just for consistency like how we have defined in other places